akka-persistence/src/test/scala/akka/persistence/SnapshotRecoveryLocalStoreSpec.scala

```diff
@@ -22,9 +22,10 @@ object SnapshotRecoveryLocalStoreSpec {
   }
 
   class LoadSnapshotTestPersistentActor(name: String, probe: ActorRef) extends NamedPersistentActor(name)
-    with TurnOffRecoverOnStart
     with ActorLogging {
 
+    override def recovery = Recovery(toSequenceNr = 0)
```

Comment:
Hm, digging into it it seems that the snapshot in the test is stored for metadata: 

```
< criteria = SnapshotSelectionCriteria(0 /* seq nr */,9223372036854775807,0,0)
> datas = Vector(SnapshotMetadata(europe,0 /* seq nr */,1496058508027))
```

so the "max replay to sequence nr = 0" actually matched that one and it would apply it.
This seems accidental though, should the max always have been -1 instead? And now we disable explicitly, so it does not replay this snapshot, which seems correct - WDYT?